components/movementsensor/adxl345/adxl345.go

```diff
@@ -41,9 +41,30 @@ var modelName = resource.NewDefaultModel("accel-adxl345")
 
 // AttrConfig is a description of how to find an ADXL345 accelerometer on the robot.
 type AttrConfig struct {
-	BoardName              string `json:"board"`
-	I2cBus                 string `json:"i2c_bus"`
-	UseAlternateI2CAddress bool   `json:"use_alternate_i2c_address,omitempty"`
+	BoardName              string              `json:"board"`
+	I2cBus                 string              `json:"i2c_bus"`
+	UseAlternateI2CAddress bool                `json:"use_alternate_i2c_address,omitempty"`
+	SingleTap              *TapAttrConfig      `json:"single_tap,omitempty"`
+	FreeFall               *FreeFallAttrConfig `json:"free_fall,omitempty"`
+}
+
+// TapAttrConfig is a description of the configs for tap registers.
+type TapAttrConfig struct {
+	AccelerometerPin int    `json:"accelerometer_pin"`
+	EchoInterrupt    string `json:"interrupt_pin"`
+	ExcludeTapX      bool   `json:"exclude_tap_x,omitempty"`
+	ExcludeTapY      bool   `json:"exclude_tap_y,omitempty"`
+	ExcludeTapZ      bool   `json:"exclude_tap_z,omitempty"`
+	ThreshTap        byte   `json:"thresh_tap,omitempty"`
+	Dur              byte   `json:"dur,omitempty"`
+}
+
+// FreeFallAttrConfig is a description of the configs for free fall registers.
+type FreeFallAttrConfig struct {
+	AccelerometerPin int    `json:"accelerometer_pin"`
+	EchoInterrupt    string `json:"interrupt_pin"`
+	ThreshFF         byte   `json:"thresh_ff,omitempty"`
+	TimeFF           byte   `json:"time_ff,omitempty"`
```

Comment:
I'd also encourage you to set these as defaults if possible. Time for Freefall would be longer than tap. But I think we can get away with:
```
type TapAttrConfig struct {
	AccelerometerPin int    `json:"accelerometer_pin,omitempty"`
        EchoInterrupt string `json:"interrupt_pin,omitempty"`
        Mode  struct `json:mode_attributes,omitempty",  //defaults to none, options are freefall, tap, all (? can you poll both modes at the same time?).

```

You can make the Mode a string, it's just that then expanding this config gets messy.